app/views/contexts/summary/block.py

```diff
@@ -12,15 +13,18 @@ def __init__(
         list_store,
         metadata,
         response_metadata,
-        schema,
+        schema: QuestionnaireSchema,
         location,
         return_to,
     ):
         self.id = block_schema["id"]
         self.location = location
         self.title = block_schema.get("title")
         self.number = block_schema.get("number")
-        self.link = self._build_link(block_schema["id"], return_to)
+        first_answer_id_for_block = schema.get_answer_ids_for_block(self.id)[0]
+        self.link = self._build_link(
+            block_schema["id"], return_to, first_answer_id_for_block
+        )
```

Comment:
Actually, this won't be quite right when questions have more than one answer. The `_build_link` method needs to be moved to `app/views/contexts/summary/answer.py`, as the link needs to depend on the specific answer. 

To cover this in a test, you just need to add two answers to one block and ensure, both change links have the appropriate answer ID for anchors/query params.